service/src/main/java/tds/config/repositories/impl/AssessmentWindowQueryRepositoryImpl.java

```diff
@@ -0,0 +1,227 @@
+package tds.config.repositories.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import tds.common.data.mapping.ResultSetMapperUtility;
+import tds.config.AssessmentWindow;
+import tds.config.model.AssessmentFormWindowProperties;
+import tds.config.repositories.AssessmentWindowQueryRepository;
+
+import static tds.common.data.mapping.ResultSetMapperUtility.mapTimeStampToInstant;
+
+@Repository
+class AssessmentWindowQueryRepositoryImpl implements AssessmentWindowQueryRepository {
+    private static final Logger LOG = LoggerFactory.getLogger(AssessmentWindowQueryRepositoryImpl.class);
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public AssessmentWindowQueryRepositoryImpl(final DataSource dataSource) {
+        jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @Override
+    public List<AssessmentWindow> findCurrentAssessmentWindows(String clientName, String assessmentId, int shiftWindowStart, int shiftWindowEnd, int sessionType) {
+        final MapSqlParameterSource parameters = new MapSqlParameterSource("clientName", clientName)
+            .addValue("assessmentId", assessmentId)
+            .addValue("sessionType", sessionType)
+            .addValue("shiftWindowStart", shiftWindowStart)
+            .addValue("shiftWindowEnd", shiftWindowEnd);
+
+        String SQL = "SELECT \n" +
+            "      DISTINCT W.numopps AS windowMax, \n" +
+            "      W.windowID,\n" +
+            "      CASE WHEN W.startDate IS NULL THEN NOW() ELSE ( W.startDate + INTERVAL :shiftWindowStart DAY) END AS startDate,\n" +
+            "      CASE WHEN W.endDate IS NULL THEN NOW() ELSE ( W.endDate + INTERVAL :shiftWindowEnd DAY) END AS endDate,\n" +
```

Comment:
We should use `UTC_TIMESTAMP()` instead of `NOW()` so that the DB timezone doesn't have to be set to UTC